lib/torrent.js

```diff
@@ -269,6 +271,38 @@ Torrent.prototype._onParsedTorrent = function (parsedTorrent) {
       self._onListening()
     })
   }
+
+  if (parsedTorrent.xs && !self.info) {
+    // TODO: Could try multiple in parallel here
+    var src = parsedTorrent.xs instanceof Array ? parsedTorrent.xs[0] : parsedTorrent.xs
+    var isHTTP = src.substring(0, 7) === 'http://' || src.substring(0, 8) === 'https://'
+
+    if (isHTTP) {
+      var opts = {
+        url: src,
+        method: 'GET',
+        headers: {
+          'user-agent': 'WebTorrent (http://webtorrent.io)'
+        }
+      }
+      this._xsRequest = get.concat(opts, function (err, res, torrent) {
+        if (err || res.statusCode !== 200 || !torrent.length) return
```

Comment:
No need to check for `!torrent.length` since that will make `parseTorrent` throw below anyway.